jax/lax_linalg.py

```diff
@@ -179,28 +178,53 @@ def eig_abstract_eval(operand):
     w = ShapedArray(batch_dims + (n,), dtype)
   else:
     raise NotImplementedError
-  return w, vl, vr
+
+  if jobvl and jobvr:
+    return w, vl, vr
+  elif jobvl:
+    return w, vl
+  elif jobvr:
+    return w, vr
+  else:
+    return w,
 
 _cpu_geev = lapack.geev
 
-def eig_cpu_translation_rule(c, operand):
+def eig_cpu_translation_rule(c, operand, jobvl, jobvr):
   shape = c.get_shape(operand)
   batch_dims = shape.dimensions()[:-2]
-  w, vl, vr, info = _cpu_geev(c, operand)
+
+  w, vl, vr, info = _cpu_geev(c, operand, jobvl=jobvl, jobvr=jobvr)
+
   ok = xops.Eq(info, xops.ConstantLiteral(c, np.array(0, np.int32)))
   w = _broadcasting_select(c, xops.Reshape(ok, batch_dims + (1,)), w,
                            _nan_like(c, w))
-  vl = _broadcasting_select(c, xops.Reshape(ok, batch_dims + (1, 1)), vl,
-                            _nan_like(c, vl))
-  vr = _broadcasting_select(c, xops.Reshape(ok, batch_dims + (1, 1)), vr,
-                            _nan_like(c, vr))
-  return xops.Tuple(c, [w, vl, vr])
+  output = [w]
+
+  if jobvl:
+    vl = _broadcasting_select(c, xops.Reshape(ok, batch_dims + (1, 1)), vl,
+                              _nan_like(c, vl))
+    output.append(vl)
 
-def eig_batching_rule(batched_args, batch_dims):
+  if jobvr:
+    vr = _broadcasting_select(c, xops.Reshape(ok, batch_dims + (1, 1)), vr,
+                              _nan_like(c, vr))
+    output.append(vr)
+
+  return xops.Tuple(c, output)
+
+def eig_batching_rule(batched_args, batch_dims, jobvl=False, jobvr=False):
   x, = batched_args
   bd, = batch_dims
   x = batching.moveaxis(x, bd, 0)
-  return eig_p.bind(x), (0, 0, 0)
+
+  output_shape_descriptor = [0]
```

Comment:
Again I wonder if this can be written more succinctly. Here's one way, although depending on your feelings about typing you might not like the reliance on Python's bool -> int promotion:
`(0,) * (1 + jobvl + jobvr)`

Reply:
On George's advice, I'm only dealing with `lapack.pyx` in this PR. The other points will be addressed in a further PR.